### PR TITLE
Pal 650 resource service redis persistence

### PR DESCRIPTION
**Pull request by dev930018**

### CODE REVIEW

File: resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/ResourceComponentTest.java [hunk 1]
```diff
@@ -50,8 +49,8 @@
 @SpringBootTest(classes = ResourceApplication.class, webEnvironment = WebEnvironment.DEFINED_PORT)
 @EnableJpaRepositories(basePackages = {"uk.gov.gchq.palisade.service.resource.repository"})
 @DirtiesContext(classMode = ClassMode.AFTER_CLASS)
+@ActiveProfiles("h2")
```
Comment (pd104923) [inline resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/ResourceComponentTest.java:52]: Why are we using h2 in tests when this is not used for the actual service?

**Review by pd104923 — CHANGES_REQUESTED** [review]

File: resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/ResourceComponentTest.java (hunk 1 above)
Comment (dev930018) [reply resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/ResourceComponentTest.java:52]: Previously h2 was default.
Can now choose between h2 and redis, h2 should be faster to spin up for small tests

File: resource-service-integration/src/test/resources/application.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2020 Crown Copyright
+# Copyright 2019 Crown Copyright
```
[the remark this answers is not in the document]
Comment (developer01189998819991197253) [reply resource-service-integration/src/test/resources/application.yaml:1]: reverted

File: resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/ResourceComponentTest.java (hunk 1 above)
Comment (kg981167224) [reply resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/ResourceComponentTest.java:52]: I would also question if we want to keep H2? Happy if that's what we want to do, probably just worth asking the question if there is no prod use for it.

If startup time is an usse we could look at https://github.com/caryyu/spring-embedded-redis-server?

File: resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/RedisPersistenceTest.java [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.junit.Assert.assertThat;
```
Comment (developer01189998819991197253) [inline resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/RedisPersistenceTest.java:54]: new tests should be written in Junit5, using AssertJ

File: resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/ScenarioPersistenceTest.java [hunk 4]
```diff
@@ -1,291 +1,293 @@
-/*
```
Comment (kg981167224) [inline resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/ScenarioPersistenceTest.java:1]: Diff has made a bit of a hash of this file, can't easily see the changes.

Comment (developer01189998819991197253) [inline resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/ScenarioPersistenceTest.java:1]: Why are the majority of changes here just deleting the contents and replacing them again?

File: resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/config/ResourceTestConfiguration.java [hunk 5]
```diff
@@ -1,36 +1,36 @@
-/*
```
Comment (developer01189998819991197253) [inline resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/config/ResourceTestConfiguration.java:1]: This file has no changes in

File: resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/ScenarioPersistenceTest.java [hunk 6]
```diff
@@ -1,291 +1,293 @@
...
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.fail;
+
+@EnableFeignClients
+@RunWith(SpringRunner.class)
+@Import(ResourceTestConfiguration.class)
+@SpringBootTest(classes = ResourceApplication.class, webEnvironment = WebEnvironment.DEFINED_PORT)
+@EnableJpaRepositories(basePackages = {"uk.gov.gchq.palisade.service.resource.repository"})
+@DirtiesContext(classMode = ClassMode.AFTER_CLASS)
+@ActiveProfiles("h2")
```
Comment (developer01189998819991197253) [inline resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/ScenarioPersistenceTest.java:64]: This seems to be the only addition to this file, why is the whole thing marked as deleted and readded??

**Review by kg981167224 — DISMISSED** [review]

File: resource-service-integration/README.md [hunk 7]
```diff
@@ -1,27 +1,27 @@
-<!---
```
Comment (developer01189998819991197253) [inline resource-service-integration/README.md:1]: There have been no changes made to this file

**Review by developer01189998819991197253 — CHANGES_REQUESTED** [review]
A bunch of files have no changes
New tests should be written in Junit5

File: resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/ResourceComponentTest.java (hunk 1 above)
Comment (dev930018) [reply resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/ResourceComponentTest.java:52]: We're already using embedded-redis for the redis tests.
I think it's just a case of how we're going to support a JVM-only deployment, if at all...
Obviously kafka will always be needed, but it at least makes sense to keep the H2 support here for now, until we decide if we want to drop it later.

File: resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/ScenarioPersistenceTest.java (hunk 6 above)
Comment (dev930018) [reply resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/ScenarioPersistenceTest.java:64]: Line endings

File: resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/ScenarioPersistenceTest.java (hunk 4 above)
Comment (dev930018) [reply resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/ScenarioPersistenceTest.java:1]: Line endings

File: resource-service-integration/README.md (hunk 7 above)
Comment (dev930018) [reply resource-service-integration/README.md:1]: Line endings

File: resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/config/ResourceTestConfiguration.java (hunk 5 above)
Comment (dev930018) [reply resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/config/ResourceTestConfiguration.java:1]: Line endings

**Review by developer01189998819991197253 — DISMISSED** [review]

File: resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/ResourceComponentTest.java (hunk 1 above)
Comment (ac74475) [reply resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/ResourceComponentTest.java:52]: I agree lets keep H2 in the tests for now, but we may loop back round to the broader question of whether we really need H2 for the JVM only deployments. At the moment it isn't doing any harm having it in

File: resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/ScenarioPersistenceTest.java [hunk 8]
```diff
@@ -1,291 +1,290 @@
...
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.fail;
+
+@EnableFeignClients
+@Import(ResourceTestConfiguration.class)
+@SpringBootTest(classes = ResourceApplication.class, webEnvironment = WebEnvironment.DEFINED_PORT)
+@EnableJpaRepositories(basePackages = {"uk.gov.gchq.palisade.service.resource.repository"})
+@DirtiesContext(classMode = ClassMode.AFTER_CLASS)
+@ActiveProfiles("h2")
```
Comment (ac74475) [inline resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/ScenarioPersistenceTest.java:61]: I would have expected a similar set of tests to these but targeting the redis active profile, the same as we do for the unit and configuration tests

**Review by ac74475 — CHANGES_REQUESTED** [review]

File: resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/ScenarioPersistenceTest.java (hunk 8 above)
Comment (developer01189998819991197253) [reply resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/ScenarioPersistenceTest.java:61]: Theres a RedisPersistenceTest which targets redis, do you require further tests?

**Review by ac74475 — APPROVED** [review]

File: resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/ScenarioPersistenceTest.java (hunk 8 above)
Comment (ac74475) [reply resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/ScenarioPersistenceTest.java:61]: nope happy now I have realised the differences

**Pull request merged**